src/common/utils/index.ts

```diff
@@ -68,3 +68,16 @@ export const generateBase64Secret = async (byteLength: number): Promise<string>
 
   return buf.toString('base64');
 };
+
+type ObjectType = Record<PropertyKey, unknown>;
+
+type PickByValue<T, V> = Pick<T, { [K in keyof T]: T[K] extends V ? K : never }[keyof T]>;
+
+type ObjectEntries<T> = { [K in keyof T]: [keyof PickByValue<T, T[K]>, T[K]] }[keyof T][];
+
+/**
+ * A helper function to get typed Object.entries
+ */
+export const getTypeSafeObjectEntries = <T extends ObjectType>(obj: T): ObjectEntries<T> => {
```

Comment:
(nit, as in optional)

I'd prefer
```suggestion
export const getTypedObjectEntries = <T extends ObjectType>(obj: T): ObjectEntries<T> => {
```